packages/flutter_tools/gradle/src/main/kotlin/FlutterPluginUtils.kt

```diff
@@ -89,6 +97,21 @@ object FlutterPluginUtils {
     @JvmName("formatPlatformString")
     fun formatPlatformString(platform: String): String = FlutterPluginConstants.PLATFORM_ARCH_MAP[platform]!!.replace("-", "_")
 
+    @JvmStatic
+    @JvmName("readPropertiesIfExist")
+    internal fun readPropertiesIfExist(propertiesFile: File): Properties {
+        val result = Properties()
+        if (propertiesFile.exists()) {
+            propertiesFile
+                .reader(StandardCharsets.UTF_8)
+                .use { reader ->
+                    // Use Kotlin's reader with UTF-8 and 'use' for auto-closing
```

Comment:
I think this is a semantic change. No requested change required.

Comment added in case an issue related to properties is bisected to this pr.